v2/pkg/engine/datasource/pubsub_datasource/pubsub_datasource.go

```diff
@@ -209,7 +210,7 @@ func (p *Planner) ConfigureSubscription() plan.SubscriptionConfiguration {
 		Input:     fmt.Sprintf(`{"topic":"%s"}`, p.current.topic),
 		Variables: p.variables,
 		DataSource: &SubscriptionSource{
-			pubSub: p.pubSub,
+			pubSub: p.pubSubBySourceName[p.current.config.SourceName],
 		},
```

Comment:
We probably we should have an additional check here that we have found source name in a map

If no, we should do p.Visitor.StopWithError

same relates to other panics, better have stop with error than panic

Reply:
agreed